cpp/test/io.cpp

```diff
@@ -77,6 +77,39 @@ void test_fides_function()
   writer1.write(0.0);
 }
 
+template <std::floating_point T>
+void test_vtx_reuse_mesh()
+{
+  auto mesh = std::make_shared<mesh::Mesh<T>>(
+      mesh::create_rectangle<T>(MPI_COMM_WORLD, {{{0.0, 0.0}, {1.0, 1.0}}},
+                                {22, 12}, mesh::CellType::triangle));
+
+  // Create a Basix continuous Lagrange element of degree 1
+  basix::FiniteElement e = basix::create_element<T>(
+      basix::element::family::P,
+      mesh::cell_type_to_basix_type(mesh::CellType::triangle), 1,
+      basix::element::lagrange_variant::unset,
+      basix::element::dpc_variant::unset, false);
+
+  // Create a scalar function space
+  auto V = std::make_shared<fem::FunctionSpace<T>>(
+      fem::create_functionspace(mesh, e));
+
+  // Create a finite element Function
+  auto u = std::make_shared<fem::Function<T>>(V);
+  auto v = std::make_shared<fem::Function<std::complex<T>>>(V);
+
+  std::filesystem::path f = "test_vtx_reuse_mesh" + std::to_string(sizeof(T))
+                            + ".bp";
+  io::VTXWriter<T> writer(mesh->comm(), f, {u, v}, "BPFile",
+                          io::VTXMeshPolicy::reuse);
+  writer.write(0);
+
+  std::fill(u->x()->mutable_array().begin(), u->x()->mutable_array().end(), 1);
+
+  writer.write(1);
```

Comment:
I don't know what is possible here, but there is no 'assertion' of correct functioning in this test.

Reply:
We could use `adios2` directly to verify the data, which I guess we should do more often.
If @massimiliano-leoni ports the test, I can have a look at what can be done wrt. verification.

Reply:
@jhale I was thinking the same thing while writing the test.
@jorgensd Do you mean porting to Python?

Reply:
Yes, If you port it to Python, I can see if I can cook up some adios2 reader verification code.

Reply:
Done. Or at least I think -- I'm not familiar with the framework.